openedx/features/course_experience/__init__.py

```diff
@@ -12,11 +14,87 @@
 # Namespace for course experience waffle flags.
 WAFFLE_FLAG_NAMESPACE = WaffleFlagNamespace(name='course_experience')
 
+# .. toggle_name: USE_DEFAULT_TRUE_NAMESPACE
+# .. toggle_implementation: DjangoSetting
+# .. toggle_default: False
+# .. toggle_description: When True, uses the new default_true namespace to help deprecate flag_undefined_default.
+# .. toggle_category: course_experience
+# .. toggle_use_cases: monitored_rollout
+# .. toggle_creation_date: 2020-06-30
+# .. toggle_expiration_date: 2020-07
+# .. toggle_warnings: n/a
+# .. toggle_tickets: n/a
+# .. toggle_status: supported
+_USE_DEFAULT_TRUE_NAMESPACE = 'USE_DEFAULT_TRUE_NAMESPACE'
+
+
+class DefaultTrueWaffleFlagNamespace(WaffleFlagNamespace):
+    """
+    This is a temporary class to help deprecate/remove ``flag_undefined_default``.
+
+    TODO: TNL-7061: Perform the actual clean-up required to remove these flags
+        and refactor/fix any tests that shouldn't be removed.
+
+    """
+    def _is_flag_active(self, flag_name):
+        """
+        Returns and caches whether the provided flag is active.
+
+        If the flag value is already cached in the request, it is returned.
+        If the flag doesn't exist, always returns default of True.
+        """
+        # Import is placed here to avoid model import at project startup.
+        from waffle.models import Flag
+
+        # validate arguments
+        namespaced_flag_name = self._namespaced_name(flag_name)
+        value = self._cached_flags.get(namespaced_flag_name)
+        if value is None:
+
+            # determine if the flag is undefined in waffle
+            try:
+                Flag.objects.get(name=namespaced_flag_name)
+            except Flag.DoesNotExist:
+                # default to True if not defined
+                value = True
+
+            if value is None:
+                request = crum.get_current_request()
+                if request:
+                    value = flag_is_active(request, namespaced_flag_name)
+                else:
+                    # Return the default value if not in a request context.
+                    # Same as the original implementation
+                    return True
+
+            self._cached_flags[namespaced_flag_name] = value
+        return value
+
+    def is_flag_active(self, flag_name, check_before_waffle_callback=None, flag_undefined_default=None):
+        """
+        Overrides is_flag_active if setting USE_DEFAULT_TRUE_NAMESPACE is True.
+        """
+        use_default_true_namespace = getattr(settings, _USE_DEFAULT_TRUE_NAMESPACE, False)
+        # TODO: REMOVE THIS HARDCODED SETTING
+        use_default_true_namespace = True
```

Comment:
**IMPORTANT: DO NO MERGE WITH THIS.**
This is just to unit test everything temporarily with the new code.